### PR TITLE
Getting an agent by topic failed

### PR DESCRIPTION
**Pull request by btry**
Fix getByTopic()

### CODE REVIEW

File: tests/suite-integration/PluginFlyvemdmAgent.php [hunk 1]
```diff
@@ -943,4 +943,26 @@ private function compareListOfMQTTMessages(array $expected, array $received) {
...
+      $this->boolean($agent->isNewItem())
+         ->isFalse(json_encode($_SESSION['MESSAGE_AFTER_REDIRECT'], JSON_PRETTY_PRINT));
+
+      $computer = new \Computer();
+      $computer->getFromDB($agent->getField(\Computer::getForeignKeyField()));
+      $entityId = $computer->getField(\Entity::getForeignKeyField());
+      $serial = $computer->getField('serial');
+
+      $emptyAgent = new \PluginFlyvemdmAgent();
+      $emptyAgent->getByTopic("$entityId/agent/$serial");
```
Comment (DIOHz0r) [inline tests/suite-integration/PluginFlyvemdmAgent.php]: Why is the serial used? this hasn't been implemented yet.

Comment (btry) [reply tests/suite-integration/PluginFlyvemdmAgent.php]: PluginFlyvemdmAgent::getByTopic() finds the agent which matches a MQTT topic. the topic contains the serial of the associated computer. Then this test must use t he serial of the computer to test the method.

I don't remember in deep the creation of the agent in  the tests context, but you should generate a random string. If not, then this needs to be added now.

Comment (DIOHz0r) [reply tests/suite-integration/PluginFlyvemdmAgent.php]: You're right however we should use the agent UUID and not the serial cause the serial could be duplicated and we will getting duplicated info also @rafaelje must change these topic value into the agent.

Comment (btry) [reply tests/suite-integration/PluginFlyvemdmAgent.php]: I also have in mind the use of the UUID, but cor now the code uses the serial. We cannot replace the serial by the UUID a few days before a release.

Let's plan this for 2.1 or 3.0 depending on a potential BC.

**Pull request merged**